waiting for cr

israeledrey opened the pull request.

israeledrey committed:
waiting for cr

yossi99905 commented on src/App.jsx:
```diff
@@ -0,0 +1,26 @@
+            <Route path="/tasks" element={<TasksList />} />
+            <Route path="/table" element={<TasksTable />} />
+            <Route path='/map' element={<TasksMap />} />
+          </Routes>
```
Move the routes to different file

israeledrey replied:
fixed

yossi99905 commented on src/components/taskDealog/SelcetDateForTask.jsx:
```diff
@@ -0,0 +1,28 @@
+
+
+
+ const SelectDateForTask = ({ func }) => {
```
Change the component name and prop name.

israeledrey replied:
fixed

yossi99905 commented on src/components/taskDealog/SelcetDateForTask.jsx:
```diff
@@ -0,0 +1,28 @@
+
+ const SelectDateForTask = ({ func }) => {
+  
+  const handleDateChange = (newValue) => {
```
Rename

israeledrey replied:
fixed

yossi99905 commented on src/components/TaskDealog.jsx:
```diff
@@ -0,0 +1,156 @@
+
+
+
+const TaskDealog = ({ showAddtPopUp, onClose, task }) => {
```
Why isn't this file in the taskDealog folder?
And after you create the task folder, you can name the file just dialog

israeledrey replied:
fixed

yossi99905 commented on src/components/TaskDealog.jsx:
```diff
@@ -0,0 +1,156 @@
+
+
+
+const TaskDealog = ({ showAddtPopUp, onClose, task }) => {
```
Naming 'showAddtPopUp'

israeledrey replied:
fixed

yossi99905 commented on src/server/api.js:
```diff
@@ -0,0 +1,48 @@
+// Render all tsks to the client
+export const getAllTasks = async () => {
+  try {
+    const response = await axios.get(`${API_URL}/tasksList`);
```
Destructuring response {data}

israeledrey replied:
fixed

yossi99905 commented on src/providers/TasksContext.jsx:
```diff
@@ -0,0 +1,60 @@
+        const fetchTasks = async () => {
+            try {
+                const tasks = await getAllTasks();
+                if (JSON.stringify(tasks) !== JSON.stringify(tasksList)) {
```
Why?

israeledrey replied:
fixed

israeledrey committed:
fix the inpinity loop in fatch req

yossi99905 commented on src/components/NavBar.jsx:
```diff
@@ -0,0 +1,83 @@
+            open={Boolean(anchorEl)}
+            onClose={handleCloseMenu}
+          >
+            <MenuItem onClick={() => handleNavigate("/map")}>📍 Map</MenuItem>
+            <MenuItem onClick={() => handleNavigate("/tasks")}>✅ Tasks</MenuItem>
+            <MenuItem onClick={() => handleNavigate("/table")}>📊 Table</MenuItem>
```
Create one function with prop

israeledrey replied:
fixed

yossi99905 commented on src/components/NavMenu.jsx:
```diff
@@ -0,0 +1,57 @@
+});
+
+
+const Navigation = () => {
```
Why have different names to the component and file?

israeledrey replied:
fixed

yossi99905 commented on src/components/SearchTask.jsx:
```diff
@@ -0,0 +1,92 @@
+    const { tasksList } = useTasksContext();
+    const [searchInput, setSearchInput] = useState("");
+
+    const InputChange = (event) => {
```
Change to somthig like onChange

israeledrey replied:
fixed

yossi99905 commented on src/pages/TasksList.jsx:
```diff
@@ -0,0 +1,46 @@
+
+
+    return (
+        <div>
```
<>

israeledrey replied:
fixed

yossi99905 commented on src/pages/TasksList.jsx:
```diff
@@ -0,0 +1,46 @@
+        <div>
+
+            <NavBar setFilteredTasks={setFilteredTasks} />
+            <div style={{ marginTop: "150px" }}>
```
Don't use in this style option

israeledrey replied:
fixed

yossi99905 commented on src/pages/TasksList.jsx:
```diff
@@ -0,0 +1,46 @@
+
+            <NavBar setFilteredTasks={setFilteredTasks} />
+            <div style={{ marginTop: "150px" }}>
+                {filteredTasks.length > 0
```
If filterTasks equals zero, then it is already false

israeledrey replied:
fixed

yossi99905 commented on src/pages/TasksList.jsx:
```diff
@@ -0,0 +1,46 @@
+                    ? (
+                        <>
+                            <p style={{ fontSize: "25px" }}>Tasks List:</p>
+                            {filteredTasks.map((task, index) => (
```
To make it more readable, put it in a function

israeledrey replied:
fixed

yossi99905 commented on src/pages/TasksList.jsx:
```diff
@@ -0,0 +1,46 @@
+                    ? (
+                        <>
+                            <p style={{ fontSize: "25px" }}>Tasks List:</p>
+                            {filteredTasks.map((task, index) => (
```
Read about useMemo

israeledrey replied:
fixed

yossi99905 commented on src/pages/TasksList.jsx:
```diff
@@ -0,0 +1,46 @@
+            </div>
+            <AddTask setShowPopup={setShowPopup} />
+
+            {showPopup && <TaskDealog showAddtPopUp={showPopup} onClose={closePopup} />}
```
Naming, For spelling mistakes, use Spelling Checker.

israeledrey replied:
fixed

yossi99905 commented on src/components/taskAction/AddTask.jsx:
```diff
@@ -0,0 +1,25 @@
+                color="primary"
+                aria-label="add"
+                sx={{ position: "fixed", bottom: 16, left: 16, zIndex: 1000 }}
+                onClick={() => setShowPopup(true)}
```
Extract a function

israeledrey replied:
fixed

yossi99905 commented on src/components/TaskDealog.jsx:
```diff
@@ -0,0 +1,156 @@
+
+
+
+const TaskDealog = ({ showAddtPopUp, onClose, task }) => {
```
Dialog

israeledrey replied:
fixed

yossi99905 commented on src/providers/TasksContext.jsx:
```diff
@@ -0,0 +1,60 @@
+        completed: false,
+        location: []
+    };
+    const [formState, setFormState] = useState(initialFormState);
```
Read about useFormik and use it

israeledrey replied:
fixed

yossi99905 commented on src/providers/TasksContext.jsx:
```diff
@@ -0,0 +1,60 @@
+                const tasks = await getAllTasks();
+                    setTasksList(tasks);
+                    setFilteredTasks(tasks);
+                    console.log("fff");
```
?

israeledrey replied:
fixed

yossi99905 commented on src/providers/TasksContext.jsx:
```diff
@@ -0,0 +1,60 @@
+            try {
+                const tasks = await getAllTasks();
+                    setTasksList(tasks);
+                    setFilteredTasks(tasks);
```
```suggestion
               setFilteredTasks(tasks);
```

israeledrey replied:
fixed

yossi99905 commented on src/providers/TasksContext.jsx:
```diff
@@ -0,0 +1,60 @@
+        const fetchTasks = async () => {
+            try {
+                const tasks = await getAllTasks();
+                    setTasksList(tasks);
```
Why 'list'?
Try to simplify the names as much as possible.

israeledrey replied:
fixed

yossi99905 commented on src/server/api.js:
```diff
@@ -0,0 +1,48 @@
+
+// Delete task
+export const deleteTask = async (_id) => {
+  if (!_id) throw new Error("Task ID is missing");
```
The schema should capture this, use in schema also in react

israeledrey replied:
fixed

yossi99905 commented on src/hooks/useTaskActions.js:
```diff
@@ -0,0 +1,35 @@
+
+    const handleAddTask = async () => {
+        const newTask = await addNewTask(formState);
+        setTasksList(prevTasks => [...prevTasks, newTask]);
+        resetFormState();
```
Note that even though you are only updating one task, whenever you change  statefrom, so getTasks happens again.

israeledrey replied:
fixed

yossi99905 commented on src/hooks/useTaskActions.js:
```diff
@@ -0,0 +1,35 @@
+
+    const handleEditTask = async (task) => {
+        const updatedTask = await updateTask(task._id, formState);
+        if (!updatedTask) {
+            console.error("Failed to update task");
+            return;
```
If it is for the user, then use it in the MUI Snackbar.

israeledrey replied:
fixed

yossi99905 commented on src/hooks/useTaskActions.js:
```diff
@@ -0,0 +1,35 @@
+            return;
+        }
+        setTasksList((prevList) =>
+            prevList.map((t) => (t._id === updatedTask._id ? updatedTask : t))
```
Instead of (t), use task to make it more readable.

israeledrey replied:
fixed

yossi99905 commented on src/hooks/useTaskActions.js:
```diff
@@ -0,0 +1,35 @@
+    const handleDeleteTask = async (_id) => {
+        try {
+            await deleteTask(_id);
+            const updatedTasks = await getAllTasks();
```
You are already writing the logic for getAll in another file.
Why don't you just remove only the deleted task like you did before?

israeledrey replied:
fixed

yossi99905 commented on src/hooks/useTaskActions.js:
```diff
@@ -0,0 +1,35 @@
+import { useTasksContext } from '../providers/TasksContext';
+import { addNewTask, updateTask, deleteTask, getAllTasks } from '../server/api';
+
+export const useTaskActions = () => {
```
Update the user about success and failure using the MUI Snackbar.

israeledrey replied:
fixed

yossi99905 replied:
What about failure, use in onerror

israeledrey replied:
fixed

yossi99905 commented on src/components/TaskDealog.jsx:
```diff
@@ -0,0 +1,156 @@
+            <FormLabel>Task Name</FormLabel>
+            <Input
+              value={formState.taskName || ""}
+              onChange={(e) => handleInputChange("taskName", e.target.value)}
```
Extract a function, and the same to all functions.

israeledrey replied:
fixed

yossi99905 commented on src/components/TaskDealog.jsx:
```diff
@@ -0,0 +1,156 @@
+
+
+  useEffect(() => {
+    if (isEditing && task) {
```
Use if and return to improve readability

israeledrey replied:
fixed

yossi99905 commented on src/components/TaskDealog.jsx:
```diff
@@ -0,0 +1,156 @@
+
+          <FormControl sx={{ gridColumn: '1/-1', width: '100%' }}>
+            <FormLabel>Task Name</FormLabel>
+            <Input
```
Make generic components for all dialog component, like in our system.

israeledrey replied:
fixed

yossi99905 commented on src/providers/TasksContext.jsx:
```diff
@@ -0,0 +1,60 @@
+export const TasksContext = createContext();
+
+
+export const TasksProvider = ({ children }) => {
```
Read about Jotai state management and use it.

israeledrey replied:
fixed

yossi99905 commented on src/providers/TasksContext.jsx:
```diff
@@ -0,0 +1,60 @@
+    const [filteredTasks, setFilteredTasks] = useState(tasksList);
+    const [isEditing, setIsEditing] = useState(false);
+
+    useEffect(() => {
```
Read about React Query and use it

yossi99905 commented on src/components/taskDealog/TaskPrioritySlider.jsx:
```diff
@@ -0,0 +1,73 @@
+
+
+
+const TaskPrioritySlider = ({fun}) => {
```
Rename fun

israeledrey replied:
fixed

yossi99905 commented on src/pages/TasksTable.jsx:
```diff
@@ -0,0 +1,141 @@
+    }
+  ], []);
+
+  const [sorting, setSortng] = useState([]);
```
To spell

israeledrey replied:
fixed

yossi99905 commented on src/pages/TasksTable.jsx:
```diff
@@ -0,0 +1,141 @@
+    onSortingChange: setSortng,
+  });
+
+  if (!tasksList || tasksList.length === 0) {
```
Same thing

israeledrey replied:
fixed

yossi99905 commented on src/pages/TasksTable.jsx:
```diff
@@ -0,0 +1,141 @@
+        <Table>
+
+          <TableHead>
+            {table.getHeaderGroups().map(headerGroup => (
```
use memo, and Extract a function

israeledrey replied:
fixed

yossi99905 commented on src/pages/TasksTable.jsx:
```diff
@@ -0,0 +1,141 @@
+  const { tasksList, filteredTasks, setFilteredTasks } = useTasksContext();
+
+
+  const data = useMemo(() => filteredTasks, [filteredTasks]);
```
Doing nothing

israeledrey replied:
fixed

yossi99905 commented on src/pages/TasksTable.jsx:
```diff
@@ -0,0 +1,141 @@
+
+  const data = useMemo(() => filteredTasks, [filteredTasks]);
+  const columns = useMemo(() => [
+    { header: "Task Name ", accessorKey: "taskName" },
```
Extract Table Header Rendering to a Separate Component, and also the rest

israeledrey replied:
fixed

yossi99905 reviewed (commented):
So far, nice job!
All the notes I gave you apply not only to the current code but to the entire project.

After u fix the code, the key areas to focus on and implement are:
Jotai for state management
React Query for data fetching and caching
Formik for form handling

I haven’t started reviewing the map-related code yet, I’ll do that after these fixes are implemented.

israeledrey committed:
waiting for cr #2

israeledrey committed:
change the route tasksList

yossi99905 commented on src/hooks/useFetchTasks.js:
```diff
@@ -0,0 +1,21 @@
+
+    const { data, isLoading, isError } = useQuery({
+        queryKey: ['tasks'],
+        queryFn: async () => {
```
Extract it into a separate function

israeledrey replied:
fixed

yossi99905 commented on src/hooks/useFetchTasks.js:
```diff
@@ -0,0 +1,21 @@
+
+    const { data, isLoading, isError } = useQuery({
+        queryKey: ['tasks'],
+        queryFn: async () => {
```
Read about react query, and if u need async/await or react query handle with this.

israeledrey replied:
fixed

yossi99905 commented on src/hooks/useFetchTasks.js:
```diff
@@ -0,0 +1,21 @@
+            const tasks = await getAllTasks();
+            setTasksList(tasks); 
+            return tasks;
+        }
```
Write the onError function for the show in Snackbar if an error occurs.

israeledrey replied:
fixed

yossi99905 commented on src/hooks/useFetchTasks.js:
```diff
@@ -0,0 +1,21 @@
+        queryKey: ['tasks'],
+        queryFn: async () => {
+            const tasks = await getAllTasks();
+            setTasksList(tasks); 
```
Why u need to update the state if u don't have tasksList state?

israeledrey replied:
fixed

yossi99905 commented on src/pages/TasksList.jsx:
```diff
@@ -0,0 +1,69 @@
+        <>
+            <NavBar />
+            <div className={classes.tasksContainer}>
+                {filteredTasks.length ?
```
!filteredTasks == filteredTasks.length ?

israeledrey replied:
fixed

yossi99905 commented on src/pages/TasksList.jsx:
```diff
@@ -0,0 +1,69 @@
+        <>
+            <NavBar />
+            <div className={classes.tasksContainer}>
+                {filteredTasks.length ?
```
Extract it into a separate function, and use in useMemo

israeledrey replied:
fixed

yossi99905 commented on src/pages/TasksList.jsx:
```diff
@@ -0,0 +1,69 @@
+
+    return (
+        <>
+            <NavBar />
```
If the navbar is called also in the task map and table, call one time before calling it.

israeledrey replied:
fixed

yossi99905 commented on src/pages/TasksTable.jsx:
```diff
@@ -0,0 +1,59 @@
+      <NavBar setFilteredTasks={setFilteredTasks} />
+
+      {
+        filteredTasks.length ?
```
Extract it into a separate function

israeledrey replied:
fixed

yossi99905 commented on src/components/taskDialog/Dialog.jsx:
```diff
@@ -0,0 +1,134 @@
+
+const Dialog = ({ showDialog, onClose, task }) => {
+  const classes = useStyles();
+  const [isEditing, setIsEditing] = useAtom(isEditingAtom);
```
setIsEditing don't call

israeledrey replied:
fixed

yossi99905 commented on src/components/table/Container.jsx:
```diff
@@ -0,0 +1,77 @@
+                <DeleteTask task={row.original} />
+            )
+        }
+    ], []);
```
Why is empty?

israeledrey replied:
fixed

yossi99905 commented on src/server/api.js:
```diff
@@ -0,0 +1,47 @@
+import axios from 'axios';
+
+const API_URL = 'http://localhost:3000';
```
Use .env

israeledrey replied:
fixed

yossi99905 commented on src/server/api.js:
```diff
@@ -0,0 +1,47 @@
+import axios from 'axios';
+
+const API_URL = 'http://localhost:3000';
```
Use in coustum axsios to set the first url once for all the other urls.

israeledrey replied:
fixed

yossi99905 reviewed (commented):
useDebounce for filers 
change the object to by key.

israeledrey committed:
Cancellation tasksListAtom and use Directly in ReactQuery, creation useFilteredTasks instead of filteredTasksAtom

israeledrey committed:
add use-debounce, dotenv and customAxios. handle errors to show everything to client and change the tasks to be object by key

israeledrey committed:
remove commads

israeledrey committed:
make the baseMap to be generic and layers by choice

israeledrey committed:
Separation of the BaseLayer To be layer on its own and make the layer to be choiced like components (without maping)

israeledrey committed:
fixed the mapInstance props to be atom, make the task featur to be generic (for point, polygon...), clean the handleFieldChange function

israeledrey committed:
fixed the updateTask request

israeledrey committed:
move the filtered
yask to backend

israeledrey committed:
fixed the edit task dialog to be update with all current feild values

israeledrey committed:
fixed the values io edit mode
make the taskSubject to be dynamic (from db)
save tasks in localStorage
make query in filter
mapstyle in prop

israeledrey committed:
fixed the filter by query to be dynamic

israeledrey committed:
some cleanings
